pkg/cache/cache.go

```diff
@@ -148,6 +179,53 @@ func (c *Cache) newClusterQueue(cq *kueue.ClusterQueue) (*ClusterQueue, error) {
 	return cqImpl, nil
 }
 
+func (c *Cache) WaitForPodsReady(ctx context.Context) {
+	c.Lock()
+	defer c.Unlock()
+
+	log := ctrl.LoggerFrom(ctx)
+	for {
+		if c.podsReadyForAllAdmittedWorkloads(ctx) {
+			return
+		}
+		log.V(3).Info("Blocking admission as not all workloads are in the PodsReady condition")
+		select {
+		case <-ctx.Done():
+			log.V(5).Info("Context cancelled, returning for the waiting")
+			return
+		default:
+			// wait releases the lock and acquires again when awaken
+			c.podsReadyCond.Wait()
+		}
+	}
+}
+
+func (c *Cache) PodsReadyForAllAdmittedWorkloads(ctx context.Context) bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.podsReadyForAllAdmittedWorkloads(ctx)
+}
+
+func (c *Cache) podsReadyForAllAdmittedWorkloads(ctx context.Context) bool {
+	log := ctrl.LoggerFrom(ctx)
+	for _, cq := range c.clusterQueues {
+		if len(cq.WorkloadsWithoutPodsReady) > 0 {
+			log.V(3).Info("There is a cluster queue with not ready workloads", "clusterQueue", cq.Name, "workloads", cq.WorkloadsWithoutPodsReady.List())
```

Comment:
also, wrap it with `if logV := log.V(3); logV.Enabled() { }`, otherwise we pay the cost of calling `List` and serializing every time.
Or just remove the list... I don't think it will be that useful.

Reply:
yup, no longer logging it